### PR TITLE
Add DASDAE compression options

### PR DESCRIPTION
**Pull request by d-chambers**
## Description

Adds optional compression controls for DASDAE writes while keeping uncompressed output as the default. Users can now pass `compression` and/or `compression_level` through `dc.write` or `patch.io.write`; a positive `compression_level` defaults to `blosc:zstd`, and level 5 is documented as the recommended general-purpose setting.

This also updates DASDAE array writing to use compressed PyTables CArrays only for non-empty, non-scalar arrays, preserving the existing uncompressed Array fallback for scalar and zero-length arrays.

## Checklist

I have (if applicable):

- [ ] referenced the GitHub issue this PR closes.
- [x] documented the new feature with docstrings and/or appropriate doc page.
- [x] included tests. See [testing guidelines](https://dascore.org/contributing/testing.html).
- [ ] added the "ready_for_review" tag once the PR is ready to be reviewed.

## Summary by CodeRabbit

* **New Features**
  * Added optional PyTables/HDF5 compression controls for writing DASDAE files, including `compression`, `compression_level`, and `shuffle`.
* **Bug Fixes**
  * Improved write/read behavior for compressed outputs, including empty selections and scalar/0-D attribute handling.
  * Ensured unsupported scalar-array compression is safely skipped.
* **Documentation**
  * Updated the file I/O tutorial with compression guidance and a `compression_level=5` example.
* **Tests**
  * Added validation of compression filter metadata and expanded DASDAE read/scan edge-case coverage.

### CODE REVIEW

File: tests/test_io/test_dasdae/test_dasdae.py [hunk 1]
```diff
@@ -171,6 +246,22 @@ def test_scan_returns_info(self, written_dascore_v1_random, random_patch):
         for key in common_keys:
             assert info1[key] == info2[key]
 
+    def test_scan_unpacks_scalar_array_attrs(self):
```
Comment (d-chambers) [inline tests/test_io/test_dasdae/test_dasdae.py:263]: Do we really need to test this private function? Maybe it is simply unreachable and should be deleted? Please check and update accordingly.

Comment (d-chambers) [reply tests/test_io/test_dasdae/test_dasdae.py:263]: Checked this path. PyTables normalizes the scalar attr before the real scan flow reaches `_get_patch_content_from_group`, so this was coverage-only defensive code. Removed the branch and the private-function test. Verification after the change: `pre-commit run --all`, `coverage run -m pytest tests/test_io/test_dasdae/test_dasdae.py`, and `coverage report --include=dascore/io/dasdae/core.py,dascore/io/dasdae/utils.py --fail-under=100 --show-missing`.
